### PR TITLE
NXP backend: Enable Amax with new Neutron flow

### PR DESCRIPTION
**Pull request by roman-janik-nxp**
### Summary
Add tests verifying correct support for amax by the Neutron backend using the new Neutron MLIR flow. 

### Test plan
Unit tests provided.

cc @robert-kalmar

### CODE REVIEW

File: backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+        # but with `keep_dim=False` it changes the shape so it's not a noop and is therefore delegated successfully.
+        keep_dim = False
+        model = AmaxModule(dim, keep_dim)
+        assert_delegated(model, input_shape, mocker, request)
+
+    def test__channels_first__keep_dim__true(self, mocker, request):
+        # Just 1 test case to verify correct handling of the `dim`.
+        # Most cases fall into the single bit error case, and since this test uses 2 operators, the error accumulates
+        #  and the final error is larger. We cannot with 100% certainty say that the error is only caused by the single
+        #  bit errors and not related to the format. That's why only this 1 case with no errors is used.
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py:246]: It may happen that there is a two-bit error, however it is not a good idea to selectively only include test cases that pass.
I'd recommend allowing two bit error or not using `remove_quant_io_ops=True` in this test.

File: backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+
+@pytest.fixture(autouse=True)
+def reseed_model_per_test_run():
+    torch.manual_seed(23)
+    np.random.seed(23)
+
+
+class AmaxModule(torch.nn.Module):
+    def __init__(
+        self, dim: int | torch.Size | list[int] | tuple[int, ...], keepdim: bool
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py:50]: Nit: According to the documentation, `dim` can only be int, tuple of ints, and can be None. `keepdim` can be None. This is also true for `MaxPoolAmaxModule`.
I think it would be better to add two test cases that use the implicit value of `dim` and `keepdim`.

File: backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+                expected_delegated_ops={
+                    MaxPool2DWithIndices: 1,
+                    GetItem: 1,
+                    Amax: 1,
+                },
+            )
+            self._assert_neutron_ir_model_has_ops(
+                model_builder_finish_spy,
+                expected_ops=[
+                    Transpose,
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py:313]: nit: in the comment above, you say no `Transpose` ops are added. Please clarify the comment if you meant to say "no `Transpose` ops are added after `MaxPool2D`".

File: backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py (hunk 1 above)
Comment (roman-janik-nxp) [reply backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py:246]: I disagree. We allow 1 bit error for single operator. In this case we have model with 2 operators, where the error can accumulate up to 2. That's why there is this disclaimer.

File: backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py (hunk 2 above)
Comment (roman-janik-nxp) [reply backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py:50]: image
I only set the hint according what I saw in docs 🤌. It is true that [amax docs](https://docs.pytorch.org/docs/2.13/generated/torch.amax.html) states what you wrote, but it accepts types in the image. 

Adding tests for implicit values is a great idea. Actually I discovered that the op behaves differently, so converter is adjusted and tests added. I also add commit that fix it for `Amin` and `Sum` ops, the `Mean` will require new converter and will be done later.

File: backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py (hunk 3 above)
Comment (roman-janik-nxp) [reply backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py:313]: Yes, that's the message. I will change it rather to "no Transpose ops are added before ReduceMax". Similarly for other reduce ops.

File: backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+            "input_shape, dim",
+            [
+                pytest.param((2, 3, 4, 5, 6), 0, id="dim=0, 5D->4D"),
+                pytest.param((2, 3, 4, 5, 6), [-3], id="dim=[-3], 5D->4D"),
+                pytest.param((1, 2, 3, 4, 5, 6), (1, -1), id="dim=(1, -1), 6D->4D"),
+            ],
+            ids=lambda dim: f"dim={dim}",
+        )
+        def test__channels_first_output(self, mocker, request, input_shape, dim):
+            model = AmaxMaxPoolModule(dim, False)
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py:409]: nit: An explanation why is this test necessary would be helpful, similarily to the tests above.

File: backends/nxp/backend/node_format_inference.py [hunk 5]
```diff
@@ -147,7 +149,7 @@ def _infer_format_of_nodes(self, node: Node):
                     self._node_inputs[node][0], DataFormat.FORMATLESS
                 )
 
-            elif op_type in [MeanDim, Amin, SumDimIntList]:
+            elif op_type in [MeanDim, Amax, Amin, SumDimIntList]:
                 # The operator schema is:
                 #  <reduce_op>(Tensor self, int[1]? dim, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
```
Comment (novak-vaclav) [inline backends/nxp/backend/node_format_inference.py]: nit: `Amax` does not have this signature, it lacks the `*, ScalarType? dtype=None`. The comment should be updated

File: backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+            )
+
+        @pytest.mark.parametrize(
+            "input_shape, dim",
+            [
+                pytest.param((2, 3, 4, 5, 6), 0, id="dim=0, 5D->4D"),
+                pytest.param((2, 3, 4, 5, 6), [-3], id="dim=[-3], 5D->4D"),
+                pytest.param((1, 2, 3, 4, 5, 6), (1, -1), id="dim=(1, -1), 6D->4D"),
+            ],
+            ids=lambda dim: f"dim={dim}",
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_amax_converter.py]: nit: this lambda `ids` is unnecessary since you already define it per test case.

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/amax_converter.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        """
+        self.assert_convertible(node)
+
+        dim, keepdim = get_reduce_node_attrs(node)
+
+        t_op = self._create_tflite_op_with_io_tensors(node)
+        t_op.builtin_options = reduce_max_options.ReduceMax(keepdim)
+
+        ops = OpsList(middle_op=t_op)
+        # dim default value is None, it that case no changes to dim or io_formats are needed and all dims are reduced
```
Comment (novak-vaclav) [inline backends/nxp/backend/ir/converter/node_converters/ops_converters/amax_converter.py]: nit: typo "it that case"

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/amin_converter.py [hunk 8]
```diff
@@ -70,7 +70,12 @@ def convert(self, node: Node):
         t_op.builtin_options = reduce_min_options.ReduceMin(keepdim)
 
         ops = OpsList(middle_op=t_op)
-        dim = get_dim_and_handle_io_formats(self.builder, ops, dim, keepdim)
+        # dim default value is None, it that case no changes to dim or io_formats are needed and all dims are reduced
```
Comment (novak-vaclav) [inline backends/nxp/backend/ir/converter/node_converters/ops_converters/amin_converter.py:73]: nit: typo "it that case"

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/sum_dim_int_list_converter.py [hunk 9]
```diff
@@ -72,7 +72,12 @@ def convert(self, node: Node):
         t_op.builtin_options = sum_options.Sum(keepdim)
 
         ops = OpsList(middle_op=t_op)
-        dim = get_dim_and_handle_io_formats(self.builder, ops, dim, keepdim)
+        # dim default value is None, it that case no changes to dim or io_formats are needed and all dims are reduced
```
Comment (novak-vaclav) [inline backends/nxp/backend/ir/converter/node_converters/ops_converters/sum_dim_int_list_converter.py:75]: nit: typo "it that case"

**Review by novak-vaclav — COMMENTED** [review]
Good job 😊 only a few nits concerning comments.

**Review by novak-vaclav — APPROVED** [review]
